File: problems/1260-shift-2d-grid/solution.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int m = grid.size();
        int n = grid[0].size();
        k = k % (m * n);
        vector<vector<int>> result(m, vector<int>(n));
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int new_index = (i * n + j + k) % (m * n);
                int new_row = new_index / n;
                int new_col = new_index % n;
                result[new_row][new_col] = grid[i][j];
            }
        }
        return result;
    }
};

File: problems/1260-shift-2d-grid/solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {
        int m = grid.length;
        int n = grid[0].length;
        k = k % (m * n);

        List<List<Integer>> shiftedGrid = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            shiftedGrid.add(new ArrayList<>());
            for (int j = 0; j < n; j++) {
                shiftedGrid.get(i).add(0);
            }
        }

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int newIndex = (i * n + j + k) % (m * n);
                int newRow = newIndex / n;
                int newCol = newIndex % n;
                shiftedGrid.get(newRow).set(newCol, grid[i][j]);
            }
        }

        return shiftedGrid;
    }
}